File: 448.FindAllNumbersDisappearedInAnArray.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        i = 0
        n = len(nums)
        result = []
        while(i<n):
            value = nums[i] - 1
            check = value >= 0 and value < n
            if check and nums[value]!=nums[i]:
                nums[value],nums[i] = nums[i],nums[value]
            else:
                i += 1
        for i in range(n):
            if nums[i]!=i+1:
                result.append(i+1)

        return result

File: .cpp/448.FindAllNumbersDisappearedInAnArray.cpp
class Solution
{
public:
    vector<int> findDisappearedNumbers(vector<int> &nums)
    {
        int n = nums.size();
        int i = 0;
        vector<int> result;
        while (i < n)
        {
            int value = nums[i] - 1;
            if (value >= 0 && value < n && nums[value] != nums[i])
            {
                swap(nums[i], nums[value]);
            }
            else
            {
                i++;
            }
        }
        for (i = 0; i < n; i++)
        {
            if (nums[i] != i + 1)
            {
                result.push_back(i + 1);
            }
        }
        return result;
    }
};